### PR TITLE
ITL-7 Add models

### PR DESCRIPTION
**Pull request by BiroAron**

### CODE REVIEW

File: server/models/todo.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import mongoose, { Model, Schema } from "mongoose";
+
+type TodoPriority = "High" | "Medium" | "Low";
+
+interface ITodo {
+  title: String;
+  priority: TodoPriority;
+  description?: String;
+  isChecked: boolean;
+  date: String;
```
Comment (tamasm-wg) [inline server/models/todo.ts]: shouldn't the date type be Date ?

File: server/models/todo.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+const TodoSchema = new Schema<ITodo>({
+  title: { type: String, required: true },
+  priority: { type: String, enum: ["High", "Medium", "Low"] },
+  description: String,
+  isChecked: Boolean,
+  date: String,
```
Comment (tamasm-wg) [inline server/models/todo.ts]: here also

File: server/models/todo.ts (hunk 1 above)
Comment (BiroAron) [reply server/models/todo.ts]: In the front-end of the application, the values of the date are set via an input field with the type of date wich gets the date in the format of dd-mm-yyyy in a string, so thats why im storing it as a string.

Comment (SzaszWebgurus) [reply server/models/todo.ts]: use Date here as type

File: server/models/todo.ts (hunk 2 above)
Comment (SzaszWebgurus) [reply server/models/todo.ts]: same as above

File: server/models/todo.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+});
+
+const todo: Model<ITodo> =
```
Comment (SzaszWebgurus) [inline server/models/todo.ts]: you dont need a variable here, you can export it here

File: server/models/user.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  password: String;
+}
+
+const UseSchema = new Schema<IUser>({
+  username: { type: String, required: true },
+  email: { type: String, required: true },
+  password: { type: String, required: true },
+});
+
+const user: Model<IUser> =
```
Comment (SzaszWebgurus) [inline server/models/user.ts]: same as above

**Review by SzaszWebgurus — CHANGES_REQUESTED** [review]

**Pull request merged**